### PR TITLE
test if we can fit a model without an intercept

### PR DESCRIPTION
**Pull request by shabbybanks**
add test for #1597 : logistic regression without intercept should not error.

### CODE REVIEW

**Comment by kevinykuo** [comment]
Is it possible to compare the fitted coefficients against `stats::glm()`?

**Comment by shabbybanks** [comment]
OK, I just added a check.

File: tests/testthat/test-ml-classification-logistic-regression.R [hunk 1]
```diff
@@ -99,6 +99,15 @@ test_that("ml_logistic_regression() agrees with stats::glm()", {
   expect_equal(unname(coef(r)), unname(coef(s)), tolerance = 1e-5)
 })
 
+test_that("ml_logistic_regression can fit without intercept",{
+  expect_error(s <- ml_logistic_regression(
+    iris_weighted_tbl,
+    formula = versicolor ~ Sepal_Width + Petal_Length + Petal_Width,
+    fit_intercept=FALSE),NA)
+  r <- glm(versicolor ~ Sepal_Width + Petal_Length + Petal_Width - 1, family=binomial(logit), data=iris_weighted)
```
Comment (javierluraschi) [inline tests/testthat/test-ml-classification-logistic-regression.R]: I believe you want to change this to:

```
r <- glm(versicolor ~ Sepal.Width + Petal.Length + Petal.Width - 1, family=binomial(logit), data=iris_weighted)
```

**Comment by shabbybanks** [comment]
yes, of course, those were wrong. fixing.

**Pull request merged**